### DISCUSSION TITLE
Setting values for multiple inputs at once

### DISCUSSION THREAD

**Discussion by vivek1996** [react-hook-form/react-hook-form#3140] [Q&A] [answered]
In my project there is a requirement to set the multiple input values depends upon the option selected by user. 
There is a autocomplete input. 
When the user selects any option some of the form fields will be pre filled. 
I cannot find the solution for this. Reset can be used but I want the fields to be dirty and validated. 

May be if setvalue accepts a object or array instead of a single value that would solve the issue 

If there is any solution already present please comment out below

**Comment by bluebill1049** [comment, marked as answer]
you will have to use setValue then, you can compose a `setValues` which contains a map with `setValue`

**Reply by vivek1996** [reply]
> you will have to use setValue then, you can compose a `setValues` which contains a map with `setValue`

Yeah, that would also work

**Reply by brayanL** [reply]
Why on version 6 isn't possible set multiple values on setValue like this?
<img width="548" alt="Screen Shot 2020-11-26 at 17 17 11" src="https://user-images.githubusercontent.com/8359783/100394500-5bed8b80-300b-11eb-81f5-802d58eae72e.png">

**Reply by bluebill1049** [reply]
hey @brayanL this syntax is too confusing, and conflict with the name of the method, which is setValue, do a `forEach` 🙏 

https://react-hook-form.com/migrate-v5-to-v6

**Reply by bluebill1049** [reply]
<img width="965" alt="Screen Shot 2020-11-27 at 9 29 40 am" src="https://user-images.githubusercontent.com/10513364/100394953-170d3780-3093-11eb-9fd0-d78e86b6a51a.png">

**Reply by brayanL** [reply]
Work like a charm thanks @bluebill1049.

**Reply by bluebill1049** [reply]
np @brayanL thanks for understanding! 🙏

**Reply by Laurene-laurene** [reply]
Salut

**Reply by akshay-prajapati** [reply]
but it's giving error for typescript file

**Reply by draysams** [reply]
> but it's giving error for typescript file

If your error is `Argument of type 'string' is not assignable to parameter of type <key of your type>` ,  you can use type assertion to enforce the type to be the key of your data type.

For instance 

``` 
[
      {
        name: 'foo' as keyof TMyDataTypeNameHere,
        value: 'bar',
      }
    ].forEach(({ name, value }) => setValue(name, value));
```

should solve the error.
